db/crud.go

```diff
@@ -895,6 +905,48 @@ func (db *Database) Put(docid string, body Body) (newRevID string, doc *Document
 			newDoc._rawBody = canonicalBytesForRevID
 		}
 
+		// Handle telling the user if there is a conflict
+		if docExistsConflict {
+			if db.ForceAPIForbiddenErrors() {
+				// Make sure the user has permission to modify the document before confirming doc existence
+				mutableBody, metaMap, newRevID, err := db.prepareSyncFn(doc, newDoc)
+				if err != nil {
+					return nil, nil, false, nil, err
+				}
+
+				_, _, _, _, _, err = db.runSyncFn(doc, mutableBody, metaMap, newRevID)
+				if err != nil {
+					err = ErrForbidden
+					base.DebugfCtx(db.Ctx, base.KeyCRUD, "Could not modify doc %q due to document exists conflict and sync func rejection: %v", base.UD(doc.ID), err)
```

Comment:
We're losing the underlying error here before we log it. These two lines need to be flipped around I think (or just return `ErrForbidden` directly)